driver-sync/src/main/com/mongodb/client/internal/ClientSessionClock.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.internal;
+
+import com.mongodb.client.Clock;
+
+public class ClientSessionClock implements Clock {
+    public static final ClientSessionClock INSTANCE = new ClientSessionClock(0L);
+
+    private long currentTime = 0L;
+
+    ClientSessionClock(final long millis) {
```

Comment:
Constructor is now private.

Reply:
I realize now that, since `ClientSessionClock` is in the same package as the only class that uses it, the class itself can be package-private rather than public.

Reply:
How will the test be able to access `ClientSessionClock` if the clock is package private and the test is not part of that package? The test is in package `com.mongodb.client` while the clock is in the package `com.mongodb.client.internal`.

Reply:
Of course, you're absolutely right.  We could move the test to the same package, but it seems more appropriate to leave it where it is.